client/payment-details/summary/missing-order-notice/index.tsx

```diff
@@ -0,0 +1,149 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { Button, RadioControl } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import Loadable from 'wcpay/components/loadable';
+import { useState } from '@wordpress/element';
+import ConfirmationModal from 'wcpay/components/confirmation-modal';
+import interpolateComponents from '@automattic/interpolate-components';
+
+/**
+ * Internal dependencies.
+ */
+
+import './style.scss';
+import CardNotice from 'wcpay/components/card-notice';
+
+interface MissingOrderNoticeProps {
+	isLoading: boolean;
+	formattedAmount: string;
+}
+
+const MissingOrderNotice: React.FC< MissingOrderNoticeProps > = ( {
+	isLoading,
+	formattedAmount,
+} ) => {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+
+	const [ reason, setReason ] = useState< string | null >( null );
+
+	const handleOnButtonClick = () => {
+		setIsModalOpen( true );
+	};
+
+	const handleModalCancel = () => {
+		setIsModalOpen( false );
+	};
+
+	const handleModalConfirmation = () => {
+		// TODO: Handle the refund.
+	};
+
+	return (
+		<>
+			<Loadable isLoading={ isLoading } placeholder="">
+				<CardNotice
+					actions={
+						<Button
+							isPrimary={ true }
+							isSmall={ false }
+							onClick={ handleOnButtonClick }
+						>
+							{ __( 'Refund', 'woocommerce-payments' ) }
+						</Button>
+					}
+				>
+					{ __(
+						'This transaction is not connected to order. Investigate this purchase and refund the transaction as needed.',
+						'woocommerce-payments'
+					) }
+				</CardNotice>
+			</Loadable>
+			{ isModalOpen && (
+				<ConfirmationModal
```

Comment:
This component supports passing class names. I'd take advantage to pass `missing-order-notice-modal`. In terms of BEM, this would be the Block.

```suggestion
				<ConfirmationModal
				    className="missing-order-notice-modal"
```